src/DataSourceProvider.vue

```diff
@@ -13,16 +13,23 @@
 
         <div v-if="dataSources.length || !dataSources.length && !selectedDataSource">
 
-          <Select2
-            :options="dataSources"
-            :selectedOption.sync="selectedDataSource"
-            :customOptionView="formatDataSourceOption"
-            :customSearch="customDataSourceSearch"
-            :optionLabelKey="'name'"
-            :optionValueKey="'id'"
-            :selectWithGroups="!!allDataSources.length"
-          >
-          </Select2>
+          <label for="data-source-select" class="select-proxy-display">
+            <select ref="select" class="hidden-select form-control" @change="onSelectChange">
```

Comment:
Why did you choose to use `@change` instead of the standard `v-model` attribute and a watcher?

Reply:
@tonytlwu Because the `this.selectedDataSource` is an object and we need it to be an object because we are looking is there any access rules set in the current data source and in native select as the value we assign a data source ID.

Reply:
@tonytlwu New [screencast](https://share.getcloudapp.com/p9uwXL9X)